tree/ntuple/v7/src/RNTupleMerger.cxx

```diff
@@ -232,32 +245,100 @@ void ROOT::Experimental::Internal::RNTupleMerger::Merge(std::span<RPageSource *>
                continue;
             }
 
+            const auto &columnDesc = descriptor->GetColumnDescriptor(columnId);
+            const auto colElement = RColumnElementBase::Generate(columnDesc.GetModel().GetType());
+
             // Now get the pages for this column in this cluster
             const auto &pages = clusterDesc.GetPageRange(columnId);
 
             RPageStorage::SealedPageSequence_t sealedPages;
+            sealedPages.resize(pages.fPageInfos.size());
+
+            const auto colRangeCompressionSettings = clusterDesc.GetColumnRange(columnId).fCompressionSettings;
+            const bool needsCompressionChange =
+               options.fCompressionSettings != -1 && colRangeCompressionSettings != options.fCompressionSettings;
+
+            // If the column range is already uncompressed we don't need to allocate any new buffer, so we don't
+            // bother reserving memory for them.
+            size_t pageBufferBaseIdx = sealedPageBuffers.size();
+            if (colRangeCompressionSettings != 0)
+               sealedPageBuffers.resize(sealedPageBuffers.size() + pages.fPageInfos.size());
 
-            std::uint64_t pageNo = 0;
             sealedPageGroups.reserve(sealedPageGroups.size() + pages.fPageInfos.size());
+
+            std::uint64_t pageNo = 0;
```

Comment:
I'd suggest using `pageNumber` or `pageIdx`. I don't think we use the `..No` naming anywhere else.